### PR TITLE
Reset groups on test contact for simulation refresh

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by norkans7** [comment]
Fixes https://github.com/rapidpro/rapidpro/issues/175 plz review

**Comment by nicpottier** [comment]
So @ewheeler and I chatted about this a bit a few weeks back. The current behavior is definitely surprising (that groups are attributes are maintained) but it does allow the user to modify the starting condition of a contact by running them through different flows. (IE, you can set up a flow that initializes the simulator contact as virgin account or as a U-Reporter) contact.

That said, I still find it confusing, and I'd rather do what you are doing here and reset the contact entirely and later add the option of having "simulator personas" that let the user explicitly set the starting conditions for a contact.

Evan, no idea when we would get around to setting up personas, but are you ok changing our behavior to clear simulator contacts in the meantime?

File: temba/flows/views.py [hunk 1]
```diff
@@ -1139,6 +1139,10 @@ def post(self, request, *args, **kwargs):
                 test_contact.name = "%s %s" % (request.user.first_name, request.user.last_name)
                 test_contact.save()
 
+                # reset the groups for test contact
+                for group in test_contact.all_groups.all():
+                    group.update_contacts([test_contact], False)
+
```
Comment (nicpottier) [inline temba/flows/views.py]: While we are at it, we should also clear all the contact fields.

Comment (norkans7) [inline temba/flows/views.py]: That is what I was about to write, sure I add that

**Comment by ewheeler** [comment]
@nicpottier @norkans7 agree -- its less confusing to clear it every time, and better to sacrifice  custom starting conditions for now.

**Comment by norkans7** [comment]
Ready

**Pull request merged**